### DISCUSSION TITLE
rPi Cam model 3 not detected

### DISCUSSION THREAD

**Discussion by f2103** [photobooth-app/photobooth-app#126] [Howtos and camera specific topics]
Hi guys,

just made a fresh install of Raspberry Pi OS from their website and followed your installation tutorial, but unfortunately the camera is not recognized. I just get a blank screen and no picture is taken.

Can you help me to troubleshoot?

-- Device --
Raspberry Pi Model 3b
Official Cam Module 3 with autofocus

-- OS --
Release date: October 10th 2023
System: 64-bit
Kernel version: 6.1
Debian version: 12 (bookworm)

-- PhotoBooth-App --
System Information
Data directory: /home/fabian/photobooth-data 
App version: 0.12.1
CPU: 1.81% / 5.85% / 5.86%
No. active threads: 7
Free disk: 50.1GB
Memory: 0.9GB total 0.1GB free 0.5GB available
Cma: 256.0MB total / 57.2MB free

Hostname: pifotobooth001
Machine: aarch64, 4 cores
Platform: Linux 6.1.0-rpi6-rpi-v8
Python executable: /home/fabian/.local/pipx/venvs/photobooth-app/bin/python
Python version: 3.11.2

CRITICAL | 13.Nov.23 18:10:14 | photobooth.routers.processing | _capture | 31 | error processing the job :\| finally failed after 3 attempts to capture image!

**Comment by mgineer85** [comment]
Here is some more information to debug. https://photobooth-app.org/support/troubleshooting/

Especially for picam pls check the ribbon cable is connected properly and 
`libcamera-hello` works correctly.

**Reply by f2103** [reply]
Hi @mgrl , 
thanks for the quick reply. `libcamera-hello` works correctly (after specifying display because i was using ssh). Unfortunately the troubleshooting guide didn't help to solve the problem.

Pls find attached the requested two log files:
[console_output.txt](https://github.com/photobooth-app/photobooth-app/files/13342947/console_output.txt)
[photobooth_log.txt](https://github.com/photobooth-app/photobooth-app/files/13342948/photobooth_log.txt)

**Reply by mgineer85** [reply]
Thank you, these are helpful information.
There seems to be something off in the underlying picamera2/libcamera installation.
You use the noir version of cm3? It should be ok but I never tested. Possible to test another camera?
Also is system updated by apt update and upgrade?

**Reply by f2103** [reply]
I almost wanted to say "yes, of course" to your question when I realized I have used `sudo apt-get update && sudo apt-get upgrade` (following the setup tutorial) instead of `sudo apt update && sudo apt upgrade`. I have absolutely no clue what the difference is, but after using 'apt' it updated libcamera and other stuff. Conclusion: IT WORKS! 💯

**Reply by mgineer85** [reply]
Great!   I will check the difference and update docs accordingly.
